src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Cookies.cs

```diff
@@ -230,7 +204,21 @@ public async Task GetAsync_SetCookieContainerAndCookieHeader_BothCookiesSent()
                     await TestHelper.WhenAllCompletedOrAnyFailed(getResponseTask, serverTask);
 
                     HttpRequestData requestData = await serverTask;
-                    string cookieHeaderValue = requestData.GetSingleHeaderValue("Cookie");
+                    string[] cookieHeaderValues = requestData.GetHeaderValues("Cookie");
+                    string cookieHeaderValue = null;
+
+                    foreach (string header in cookieHeaderValues)
+                    {
+                        if (cookieHeaderValue == null)
+                        {
+                            cookieHeaderValue = header;
+                        }
+                        else
+                        {
+                            // rfc7540 8.1.2.5 states multiple cookie headers should be represented as single value.
+                            cookieHeaderValue = String.Concat(cookieHeaderValue, "; ", header);
```

Comment:
I think this should only happen for HTTP2, right? Seems like we should ensure the correct behavior here for HTTP/1.1 and HTTP2 independently. I.e. make sure HTTP/1.1 is sending a single header, and HTTP2 is sending multiple (two? or whatever).

Reply:
common function was added.